phoenix-core/src/main/java/org/apache/phoenix/schema/transform/SystemTransformRecord.java

```diff
@@ -0,0 +1,256 @@
+        private String transformJobId;
+        private int transformRetryCount =0;
+        private Timestamp startTs = new Timestamp(EnvironmentEdgeManager.currentTimeMillis());
+        private Timestamp endTs;
+        private String oldMetadata;
+        private String newMetadata;
+        private String transformFunction;
+
+        public SystemTransformBuilder setTransformType(PTable.TransformType transformType) {
+            this.transformType = transformType;
+            return this;
+        }
+
+        public SystemTransformBuilder setSchemaName(String schemaName) {
+            this.schemaName = schemaName;
+            return this;
+        }
+
+        public SystemTransformBuilder setLogicalTableName(String tableName) {
+            this.logicalTableName = tableName;
+            return this;
+        }
+
+        public SystemTransformBuilder setTenantId(String tenant) {
+            this.tenantId = tenant;
+            return this;
+        }
+
+        public SystemTransformBuilder setLogicalParentName(String name) {
+            this.logicalParentName = name;
+            return this;
+        }
+
+        public SystemTransformBuilder setNewPhysicalTableName(String tableName) {
+            this.newPhysicalTableName = tableName;
+            return this;
+        }
+
+        public SystemTransformBuilder setTransformStatus(String transformStatus) {
+            this.transformStatus = transformStatus;
+            return this;
+        }
+
+        public SystemTransformBuilder setTransformJobId(String transformJobId) {
+            this.transformJobId = transformJobId;
+            return this;
+        }
+
+        public SystemTransformBuilder setOldMetadata(String oldMetadata) {
+            this.oldMetadata = oldMetadata;
+            return this;
+        }
+
+        public SystemTransformBuilder setNewMetadata(String newMetadata) {
+            this.newMetadata = newMetadata;
+            return this;
+        }
+
+        public SystemTransformBuilder setTransformRetryCount(int transformRetryCount) {
+            this.transformRetryCount = transformRetryCount;
+            return this;
+        }
+
+        public SystemTransformBuilder setStartTs(Timestamp startTs) {
+            this.startTs = startTs;
+            return this;
+        }
+
+        public SystemTransformBuilder setEndTs(Timestamp endTs) {
+            this.endTs = endTs;
+            return this;
+        }
+
+        public SystemTransformBuilder setTransformFunction(String transformFunction) {
+            this.transformFunction = transformFunction;
+            return this;
+        }
+
+        public SystemTransformRecord build() {
+            Timestamp end = endTs;
+            if (end == null && transformStatus != null && transformStatus.equals(PTable.TaskStatus.COMPLETED.toString())) {
+                end = new Timestamp(EnvironmentEdgeManager.currentTimeMillis());
```

Comment:
Shall we ignore `end == null` case and always set `end` to current time?

Reply:
Transform#addTransform is anyways not setting `endTs` and it looks like this condition would be useful for future purpose when transform usecases are onboarded. Let me mark this resolved.